src/gui/options_banlist.cpp

```diff
@@ -31,10 +31,20 @@
 #include "ui_options_banlist.h"
 
 #include <QMessageBox>
+#include <QHostAddress>
 #include <libtorrent/address.hpp>
 
 #include "base/bittorrent/session.h"
 
+namespace
+{
+    inline bool isValidIP(const QString &ip)
+    {
+        QHostAddress addr;
```

Comment:
I am in doubt. Your approach will return false for a "0.0.0.0" string, But my will return true. Would we like to allow user ban "0.0.0.0" or to disallow ?